src/components/Logo/LogoTable.ts

```diff
@@ -0,0 +1,164 @@
+import uriToHttp from 'utils/uriToHttp'
+
+import { getAssetsRepoURI, getNativeLogoURI } from './util'
+
+export type LogoTableInput = { address?: string | null; chainId: number; isNative?: boolean; logoURI?: string }
+
+abstract class LogoSrc {
+  abstract key: string
+  abstract getUri: () => string | undefined
+  abstract useAlternateUri?: () => string | undefined
+}
+
+class UriSrc implements LogoSrc {
+  key: string
+  protected parsedUri: string | undefined | null = null
+  protected unparsedUri: string
+  private alternateUri: string | undefined
+
+  constructor(uri: string) {
+    this.key = uri
+    this.unparsedUri = uri
+  }
+
+  getUri() {
+    // Lazy-parse the address
+    if (this.parsedUri === null) {
+      ;[this.parsedUri, this.alternateUri] = uriToHttp(this.unparsedUri)
```

Comment:
remove semicolon?

Reply:
if this is intentional , i'm not familiar with the syntax. so just checking

Reply:
the linter won't allow this w/out semi colon. Do you know if theres a cleaner way do destructure into existing vars?

Reply:
maybe just using local variables? i've never seen this syntax before but that doesn't mean it's bad